hudi-client/hudi-client-common/src/main/java/org/apache/hudi/client/BaseHoodieClient.java

```diff
@@ -262,6 +282,18 @@ protected void finalizeWrite(HoodieTable table, String instantTime, List<HoodieW
     }
   }
 
+  class GetMetadataWriterFunc implements Functions.Function2<String, HoodieTableMetaClient, Option<HoodieTableMetadataWriter>> {
```

Comment:
You can also define a FunctionalInterface to make it more explicit rather than relying on the general `Functions`